tools/src/tester/TestsRunner.ts

```diff
@@ -0,0 +1,65 @@
+import { type OpenAPIV3 } from 'openapi-types'
+import SpecParser from './SpecParser'
+import ChapterReader from './ChapterReader'
+import SchemaValidator from './SchemaValidator'
+import StoryEvaluator, { type StoryFile } from './StoryEvaluator'
+import fs from 'fs'
+import { type Story } from './types/story.types'
+import { read_yaml } from '../../helpers'
+import { type StoryEvaluation } from './types/eval.types'
+import ResultsDisplayer from './ResultsDisplayer'
+
+declare global {
+  // eslint-disable-next-line no-var
+  var chapter_reader: ChapterReader
+  // eslint-disable-next-line no-var
+  var schema_validator: SchemaValidator
+  // eslint-disable-next-line no-var
+  var spec_parser: SpecParser
+}
```

Comment:
Using globals is a code smell to me and would prefer passing instances via constructors, especially given these are being built and assigned from the constructor below.

Reply:
You could use a singleton.